SW-1352 add support for pep 440 in the mr beam plugin

khaledsherkawi opened the pull request:
SW-1352 Refactor the PEP version comparison class

SW-1352 Add frontend and backend methods to handle PEP440 comparisons via API calls

SW-1352 Fix the naming of the Mr Beam Plugin version constants and use the PEP440 comparison method

khaledsherkawi committed:
SW-1352 Refactor the PEP version comparison class

khaledsherkawi committed:
SW-1352 Add frontend and backend methods to handle PEP440 comparisons via API calls

khaledsherkawi committed:
SW-1352 Fix the naming of the Mr Beam Plugin version constants and use the PEP440 comparison method

JFSolution commented on octoprint_mrbeam/version_comparator.py:
```diff
@@ -0,0 +1,36 @@
+from octoprint_mrbeam.mrb_logger import mrb_logger
```
please move the file to the util

khaledsherkawi replied:
done

JFSolution commented on octoprint_mrbeam/__init__.py:
```diff
@@ -159,6 +161,14 @@ class MrBeamPlugin(
     TIME_NTP_SYNC_CHECK_INTERVAL_FAST = 10.0
     TIME_NTP_SYNC_CHECK_INTERVAL_SLOW = 120.0
 
+    COMPARISON_OPTIONS = [
```
please move this to the verion_comperator file

khaledsherkawi replied:
done

JFSolution commented on tests/softwareupdate/test_comparison.py:
```diff
@@ -1,20 +1,19 @@
 
-def bla(comp1, comparision_options):
-    return VersionComperator.get_comperator(comp1, comparision_options).priority
+def bla(comp1, comparison_options):
```
please rename or try to use it without this function

khaledsherkawi replied:
done

JFSolution commented on octoprint_mrbeam/__init__.py:
```diff
@@ -1306,6 +1307,21 @@ def generate_backlash_compenation_pattern_gcode(self, data):
             return jsonify(res)
 
+    # simpleApiCommand: compare_pep440_versions;
+    def compare_pep440_versions(self, data):
```
Can you please add a Google Doc string

khaledsherkawi replied:
done

JFSolution commented on octoprint_mrbeam/__init__.py:
```diff
@@ -104,6 +106,7 @@
 from octoprint_mrbeam.util.uptime import get_uptime, get_uptime_human_readable
 from octoprint_mrbeam.util import get_thread
 from octoprint_mrbeam import camera
+from .version_comparator import VersionComparator
```
please change it to octoprint_mrbeam.version.comperator

khaledsherkawi replied:
done

khaledsherkawi committed:
SW-1352 Refactor version_comparator.py and COMPARISON_OPTIONS into Util package

khaledsherkawi committed:
SW-1352 Add doc string to compare_pep440_versions method

JFSolution commented on octoprint_mrbeam/util/version_comparator.py:
```diff
@@ -34,3 +34,12 @@ def get_comparator(comparision_string, comparision_options):
                 return item
+
+
+COMPARISON_OPTIONS = [
```
I would place this at the beginning of the file

khaledsherkawi replied:
This shows an error when placed at the top. The IDE marks the class 'VersionComparator' as not defined yet

JFSolution replied:
okay

JFSolution commented on octoprint_mrbeam/software_update_information.py:
```diff
@@ -515,12 +516,12 @@ def _generate_config_of_beamos(plugin, moduleconfig, beamos_version, tierversion
 
     config_for_beamos = get_config_for_version(
-        beamos_version, sorted_config_for_beamos_versions, plugin.COMPARISON_OPTIONS
+                                                                                                                                                                                                                    beamos_version, sorted_config_for_beamos_versions, COMPARISON_OPTIONS
```
there is something messy with this line, please check

khaledsherkawi replied:
done

khaledsherkawi committed:
SW-1352 Reformat line

khaledsherkawi committed:
SW-1352 Fix API command return

As the simpleApiCommand is an AJAX call then it is by default asynchronous. This means that the return statements will be executed before the response is defined. The fix is thus by setting the async parameter to false.

JFSolution reviewed (commented):
Did you change anything in prettier? There is a lot of reformating

JFSolution commented on octoprint_mrbeam/__init__.py:
```diff
@@ -1296,6 +1300,33 @@ def generate_backlash_compenation_pattern_gcode(self, data):
+                comparator, COMPARISON_OPTIONS
+            ).compare(v1, v2)
+            return make_response(json.dumps(result), 200)
+        except Exception as e:
```
Please handle the exception or catch before if one of the parameters is missing

khaledsherkawi replied:
Done

khaledsherkawi committed:
Revert "SW-1352 Fix API command return"

This reverts commit 2682a09677017354f5b89bf454f1be47de04de0f.

khaledsherkawi committed:
SW-1352 Fix API command return

As the simpleApiCommand is an AJAX call then it is by default asynchronous. This means that the return statements will be executed before the response is defined. The fix is thus by setting the async parameter to false.

khaledsherkawi commented:
> Did you change anything in prettier? There is a lot of reformating

Done - fixed

khaledsherkawi committed:
SW-1352 Return None upon exception

khaledsherkawi committed:
SW-1352 Refactor compare_pep440_versions method

khaledsherkawi committed:
SW-1352 Add a test for compare_pep440_versions method
